contrib/to-a11y-string/test/to-a11y-string-spec.js

```diff
@@ -0,0 +1,368 @@
+        });
+    });
+
+    describe("enclose", () => {
+        test("\\cancel", () => {
+            const result = renderString("\\cancel{a}");
+            expect(result).toMatchInlineSnapshot(
+                `"start cancel, a, end cancel"`,
+            );
+        });
+
+        test("\\fbox", () => {
+            const result = renderString("\\fbox{a}");
+            expect(result).toMatchInlineSnapshot(`"start box, a, end box"`);
+        });
+
+        test("\\sout", () => {
+            const result = renderString("\\sout{a}");
+            expect(result).toMatchInlineSnapshot(`"start strikeout, a, end strikeout"`);
+        });
+    });
+
+    describe("exponents", () => {
+        test("simple exponent", () => {
+            const result = renderString("e^x");
+            expect(result).toMatchInlineSnapshot(
+                `"e, start superscript, x, end superscript"`,
+            );
+        });
+
+        test("^{\\circ} => degrees", () => {
+            const result = renderString("90^{\\circ}");
+            expect(result).toMatchInlineSnapshot(`"90, degrees"`);
+        });
+
+        test("^{\\degree} => degrees", () => {
+            const result = renderString("90^{\\degree}");
+            expect(result).toMatchInlineSnapshot(`"90, degrees"`);
+        });
+
+        test("^{\\prime} => prime", () => {
+            const result = renderString("f^{\\prime}");
+            expect(result).toMatchInlineSnapshot(`"f, prime"`);
+        });
+
+        test("^2 => squared", () => {
+            const result = renderString("x^2");
+            expect(result).toMatchInlineSnapshot(`"x, squared"`);
+        });
+
+        test("^3 => cubed", () => {
+            const result = renderString("x^3");
+            expect(result).toMatchInlineSnapshot(`"x, cubed"`);
+        });
+
+        test("log_2", () => {
+            const result = renderString("\\log_2{x+1}");
+            expect(result).toMatchInlineSnapshot(
+                `"log, start subscript, 2, end subscript, x, plus, 1"`,
+            );
+        });
+    });
+
+    describe("genfrac", () => {
+        test("simple fractions", () => {
+            const result = renderString("\\frac{2}{3}");
+            expect(result).toMatchInlineSnapshot(
+                `"start fraction, 2, divided by, 3, end fraction"`,
+            );
+        });
+
+        test("nested fractions", () => {
+            const result = renderString("\\frac{1}{1+\\frac{1}{x}}");
+            // TODO: this result is ambiguous, we need to fix this
+            expect(result).toMatchInlineSnapshot(
+                `"start fraction, 1, divided by, 1, plus, start fraction, 1, divided by, x, end fraction, end fraction"`,
+            );
+        });
+
+        test("binomials", () => {
+            const result = renderString("\\binom{n}{k}");
+            expect(result).toMatchInlineSnapshot(
+                `"start binomial, left parenthesis, n, over, k, right parenthesis, end binomial"`,
```

Comment:
I wouldn't have expected parenthesis in addition to binomial.

Reply:
Yeah, this isn't very good.  It looses all of the meaning.  I think having some way to leave things unexpanded would help with this but it would complicate ParseNode.  Also, not expanding certain things could result in an unparsable expression. :\

Reply:
I think a good solution for this might be to add macros before calling KaTeX, to customize the rendering of some commands, like we already do to make cleaner MathML outputs (check out the existing macros in `macros.js` that use `\html@mathml`).  I could imagine macro-defining `\binom` to something like `{\@speak{begin}\@speak{binomial}#1\@speak{choose}#2\@speak{end}\@speak{binomial}}` for a11y purposes.  Ideally then `\@speak` could be `defineFunction`d to create a parse node that then just gets output in a11y.

Reply:
I worry about injecting human readable strings into our macros.  It doesn't well with i18n.

Reply:
I think I get what you're saying now.  `renderA11yString` would be responsible for defining the macros.  That's a really cool idea.